app/lib/archive_deletion_belonging_to_application.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module ArchiveDeletionBelongingToApplication
+  extend ActiveSupport::Concern
+  included do
+    after_destroy :archive_as_deleted, if: :destroyed_by_association
+  end
+
+  private
+
+  def archive_as_deleted
+    ::DeletedObject.create!(object: self, owner: application, metadata: {value: value})
```

Comment:
so if `create!` would fail, it would roll back the whole transaction.. is that desired?

Reply:
Yes, IMO, it should be part of the destruction of the object, and if this fails, the destroy should fail.
Because if this fails silently, we are not cleaning it up from Apisonator.
WDYT @macejmic ?

Reply:
ok, fair enough.. but let's put it to a models/concerns.. that's the right directory for sharing logic between the models.. and maybe some other name? SaveAssociation(For)ApplicationDestroy or something like that..